cdap-master/src/main/java/co/cask/cdap/data/runtime/main/TransactionServiceTwillRunnable.java

```diff
@@ -93,6 +97,7 @@ protected Injector doInit(TwillContext context) {
   @Override
   public void addServices(List<? super Service> services) {
     services.add(injector.getInstance(TransactionService.class));
+    services.add(injector.getInstance(TransactionHttpService.class));
```

Comment:
So you are running this http service in all TX containers (both the active and all standbys), meaning even if there is no active tx service, the ping call still return ok. E.g. the active one died and the standby one is not taking over yet (since zk timeout is 20 secs and loading of snapshots and replaying WAL takes time).